### PR TITLE
Add 2D/3D visibility switch 

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 

- Add 2D/3D visibility switch component which can only be visible if the `mapbox`, `3dwalls` and `floorplan` modules are enabled in the CMS 

# How 

- Create a  `mapboxViewMode` atom which can hold three values: initial3D, clicked3D or clicked2D
- Create a `currentPitch` Recoil selector that combines the `pitch` property with the `mapboxViewMode` into a calculated value.
- Create a `ViewModeSwitch.jsx` component which holds the switch buttons and sets the mapboxViewMode when clicked. It also sets the desired pitch and turns on and off the visibility of features on the map.
- The `useMapBoundsDeterminer` uses the `currentPitch` selector to apply the correct pitch when eg. a venue changes.
- Use the `useReset` method to handle the `timeout` property

### CODE REVIEW

File: packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import { ReactComponent as Dark3D } from '../../../../assets/3d-dark.svg';
+import primaryColorState from '../../../../atoms/primaryColorState';
+import mapboxViewModeState from '../../../../atoms/mapboxViewModeState';
+import './ViewModeSwitch.scss';
+import currentPitchSelector from '../../../../selectors/currentPitch';
+
+export const ViewModes = Object.freeze({
+    'initial3D': 1,
+    'clicked3D': 2,
+    'clicked2D': 3
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx:16]:
```suggestion
    initial3D: 1,
    clicked3D: 2,
    clicked2D: 3
```

File: packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { ReactComponent as Light2D } from '../../../../assets/2d-light.svg';
+import { ReactComponent as Dark2D } from '../../../../assets/2d-dark.svg';
+import { ReactComponent as Light3D } from '../../../../assets/3d-light.svg';
+import { ReactComponent as Dark3D } from '../../../../assets/3d-dark.svg';
+import primaryColorState from '../../../../atoms/primaryColorState';
+import mapboxViewModeState from '../../../../atoms/mapboxViewModeState';
+import './ViewModeSwitch.scss';
+import currentPitchSelector from '../../../../selectors/currentPitch';
+
+export const ViewModes = Object.freeze({
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx:13]: Maybe we could add comment(s) explaining the values.

File: packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+function ViewModeSwitch({ mapView }) {
+
+    const solution = useRecoilValue(solutionState);
+    const primaryColor = useRecoilValue(primaryColorState);
+    const [viewMode, setViewMode] = useRecoilState(mapboxViewModeState);
+    const [shouldShowSwitch, setShouldShowSwitch] = useState(false);
+    const currentPitch = useRecoilValue(currentPitchSelector);
+
+    useEffect(() => {
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx:27]: Make a comment explaining what we do here.

File: packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            setViewMode(ViewModes.initial3D);
+        }
+    }, [solution]);
+
+    useEffect(() => {
+        // Toggle 2D and 3D features on the map
+        if (mapView?.isReady) {
+            // If the viewMode is null, set the ViewMode to be initial3D
+            if (viewMode === null) {
+                setViewMode(ViewModes.initial3D);
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx:39]: Could we have this as the default atom value? 🤔

File: packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                case ViewModes.initial3D:
+                    mapView.tilt(currentPitch, 2000);
+                    mapView.hideFeatures([mapView.MapboxFeatures.MODEL2D, mapView.MapboxFeatures.WALLS2D]);
+                    break;
+                case ViewModes.clicked3D:
+                    mapView.tilt(45, 2000);
+                    mapView.hideFeatures([mapView.MapboxFeatures.MODEL2D, mapView.MapboxFeatures.WALLS2D]);
+                    break;
+                default:
+                // Intentionally left blank
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx:55]:
```suggestion
                    // Intentionally left blank
```

File: packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            }
+        }
+    }, [mapView?.isReady, viewMode]);
+
+    return (
+        <>
```
Comment (ammapspeople) [inline packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx:61]: Not sure we need this.

Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx]: ✅

File: packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx (hunk 4 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx]: Yes, refactored the code. ✅

File: packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx (hunk 2 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/Map/MapboxMap/ViewmodeSwitch/ViewModeSwitch.jsx]: ✅

File: packages/map-template/src/components/MapTemplate/MapTemplate.scss [hunk 7]
```diff
@@ -30,6 +34,10 @@
         .venue-selector__button {
             display: none;
         }
+
+        .switch {
```
Comment (ammapspeople) [inline packages/map-template/src/components/MapTemplate/MapTemplate.scss:38]: The "switch" class name is a bit too generic, I think. Consider renaming it to something more specific.

Comment (andreeaceachir142) [reply packages/map-template/src/components/MapTemplate/MapTemplate.scss]: ✅

Comment (ammapspeople) [reply packages/map-template/src/components/MapTemplate/MapTemplate.scss]: 👏

**Review by ammapspeople — APPROVED** [review]
🆗

**Comment by andreeaceachir142** [comment]
Ready for review again, fixed all the issues found earlier. ✅ @ammapspeople

**Comment by ammapspeople** [comment]
@andreeaceachir142 I just realized this: we're adding a new way of adding a UI control on top of the map (let's call it technical debt).
The venue selector and 2D/3D switch visually belong to the same group of elements. Also, styling & position would be more straightforward if they actually did.
We could try to use a React portal to make that happen.
But that could be a separate "tech debt removal" task.

**Comment by ammapspeople** [comment]
Other than that, I maintain my approval ✅ 🆗

**Pull request merged**